### PR TITLE
Update ODL version

### PR DESCRIPTION
**Pull request by ElizabethOkerio**
Updated the ODL version to `8.0.0-preview.3`. Since `8.0.0-preview.3` had some breaking changes, I've also made some updates like: 

1. Getting rid of the `IContainerBuilder` and using `Microsoft.Extensions.DependencyInjection` instead. 
2. Removed references to `ODataSimplifiedOptions` and used `ODataMessageReaderSettings` instead since we got rid of this class in `ODL 8.0.0-preview.3`
3. I've also made changes to ODataError's `ErrorCode` to `Code` and the way the ODataInnerError is initialized from `new ODataInnerError { Message = ConvertModelStateErrors(modelStateError) };` to `new Dictionary<string, ODataValue>
                            {
                                {
                                    "Message", new ODataPrimitiveValue(ConvertModelStateErrors(modelStateError))
                                }
                            });`

### CODE REVIEW

File: sample/BenchmarkServer/BenchmarkServer.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>net7.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (xuzhg) [inline sample/BenchmarkServer/BenchmarkServer.csproj:4]: If we update to ODL 8, we need to update our version to 9 for ASP.NET Core OData. right?

Comment (ElizabethOkerio) [reply sample/BenchmarkServer/BenchmarkServer.csproj:4]: yes

**Review by habbes — CHANGES_REQUESTED** [review]
We shouldn't target this PR against the main branch due to potential breaking/disruptive changes.

**Comment by ElizabethOkerio** [comment]
> We shouldn't target this PR against the main branch due to potential breaking/disruptive changes.

Yes. Creating a branch for it.

**Review by habbes — COMMENTED** [review]
We should set the version to 9.0.0-preview.1

**Pull request closed without merging**